components/datetime/src/format/zoned_datetime.rs

```diff
@@ -102,12 +101,11 @@ where
         .map(|s| s.get());
 
     match field.symbol {
```

Comment:
Shane and Eric, do we still need this branch `datetime::write_field(pattern, field, symbols, loc_datetime, w)`? 

If we don't, then I am fine to remove `field`. Otherwise, we need to figure out how to merge these 2 branches first.

Reply:
@nordzilla The issue is that patterns can have multiple time zone placeholders. In the old model, when we always had all the data in TimeZoneFormat, we could wait until "format time" to pick the field. Now, we are creating a TimeZoneFormat that is specific to a certain field.

My preference is to return an error from ZonedDateTimeFormat if it is given a pattern that has multiple different time zone fields in it, and revisit this issue if a use case comes up.

Thoughts?

Reply:
Sorry, I missed this notification.

Your solution seems fine to me. I am not aware of any cases that have multiple time zone fields.

Reply:
If I remove the `field` here, I found tests `test_components_with_zones` and `test_length_fixtures` will fail. For example, the test case for `test_components_with_zones` is 
```
{
        "description": "Full date time example with long generic time zone",
        "input": {
            "value": "2020-01-21T08:25:07.000+05:00",
            "options": {
                "components": {
                    "weekday": "long",
                    "month": "long",
                    "day": "numeric",
                    "year": "numeric",
                    "hour": "two-digit",
                    "minute": "two-digit",
                    "second": "two-digit",
                    "time_zone_name": "long-generic"
                }
            }
        },
        "output": {
            "values": {
                "en": "Tuesday, January 21, 2020 at 08:25:07 GMT+05:00"
            }
        }
    }
```

And `field.symbol` is `FieldSymbol::Weekday` instead of `FieldSymbol::TimeZone`. If we believe that `field` is not used anymore, then we should update the test case. Otherwise, I will leave the code untouched.

Reply:
@samchen61661 I believe that this is resolved now that we are checking for the _same_ symbol and not just _one_ symbol, correct?

Reply:
no, I found that if I remove the `field` here, it will fail tests`test_components_with_zones` and `test_length_fixtures`. (my previous comment had all the details)

I am not familiar with the tests. But my gut is that this code is not only used by time zone component, but also used by week day component. (I guess it also uses pattern?)

Reply:
I just noticed that this is `FormattedZonedDateTime`, not `FormattedTimeZone`.